addons/mail/static/src/core/common/suggestion_hook.js

```diff
@@ -173,19 +177,23 @@ class UseSuggestion {
         if (!this.search.delimiter) {
             return;
         }
-        const { type, suggestions } = this.suggestionService.searchSuggestions(this.search, {
+        const {
+            type,
+            suggestions,
+            specialSuggestions = [],
+        } = this.suggestionService.searchSuggestions(this.search, {
             thread: this.thread,
             sort: true,
         });
-        if (!suggestions.length) {
+        if (!suggestions.length && !specialSuggestions.length) {
             this.state.items = undefined;
             return;
         }
         // arbitrary limit to avoid displaying too many elements at once
         // ideally a load more mechanism should be introduced
         const limit = 8;
         suggestions.length = Math.min(suggestions.length, limit);
-        this.state.items = { type, suggestions };
+        this.state.items = { type, suggestions: [...suggestions, ...specialSuggestions] };
```

Comment:
This diff is unnecessary with with putting `specialSuggestions` in `suggestions`. Also the `limit=8` is properly applied even with `@everyone`, which is the intended limit in all showing of suggestions